### DISCUSSION TITLE
Flexible Server Rule Azure.PostgreSQL.AAD

### DISCUSSION THREAD

**Discussion by paulmccrady** [Azure/PSRule.Rules.Azure#2743] [Q&A]
We have configured a Postgres Flexible Server and have the following config under the Security->Authentication blade.

A Microsoft Entra Admin is configured (but omitted from screenshot for security purposes).
![Screenshot 2024-03-07 at 09 56 53](https://github.com/Azure/PSRule.Rules.Azure/assets/30175621/f1034799-21a2-4a11-b754-66251071fcec)

The following rule fails:

[FAIL] Azure.PostgreSQL.AAD (AZR-000389)
| File: xxx/out/templates/xxx.json:1:2773703
| RECOMMEND:
| Consider using Azure Active Directory (AAD) authentication with Azure Database
| for PostgreSQL databases. Additionally, consider disabling PostgreSQL
| authentication.
| REASON:
| - A sub-resource of type 'Microsoft.DBforPostgreSQL/flexibleServers/administrators' has not been specified.
| HELP:
| - [https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.PostgreSQL.AAD/](https://apc01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fazure.github.io%2FPSRule.Rules.Azure%2Fen%2Frules%2FAzure.PostgreSQL.AAD%2F&data=05%7C02%7Cpaul.mccrady%40cognizant.com%7Ce6a210d0dc6f45f88d1808dc3e8ec42a%7Cde08c40719b9427d9fe8edf254300ca7%7C0%7C0%7C638454030120436883%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=ujhNQZfL8DJAif2CVOdGe6K27sqiPQxh8qWykJ%2Bccjk%3D&reserved=0)

We can see that the returned json does not contain a ‘Microsoft.DBforPostgreSQL/flexibleServers/administrators’ type.

Is this a rule issue or Export or something else?

**Comment by BernieWhite** [comment]
@paulmccrady could be, let me do some investigation.

**Comment by BernieWhite** [comment]
@paulmccrady Yes you're correct, this type doesn't have an export method. I'll raised the bug #2744 to fix this issue.
